ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -147,6 +147,11 @@ - (void)updateViewControllerIfNeeded
   // if nav is nil, it means we can be in a fullScreen modal, so there is no nextVC, but we still want to update
   if (vc != nil && (nextVC == vc || isInFullScreenModal || isPresentingVC)) {
     [RNSScreenStackHeaderConfig updateViewController:self.screenView.controller withConfig:self animated:YES];
+    // As the header might have change in `updateViewController` we need to ensure that header height
+    // returned by the `onHeaderHeightChange` event is correct.
+    if ([vc isKindOfClass:[RNSScreen class]]) {
```

Comment:
I believe this check is redundant as `screenView.controller` is always of type `RNSScreen`.

Reply:
In this case we cannot look for screenView, as this method is static 😅 Do you mean that we don't need to ensure if vc is RNSScreen?